### PR TITLE
Utm adapter upstream v1

### PR DESCRIPTION
**Pull request by rpashchapur**
UTM adapter V1 (https://github.com/Dronecode/utm-adapter)

Description
-----------
In this PR the following things are done

- CMake Build System Updated
- Boost Dependencies Removed 
- QNetworkAccessManager 
- QML Resources Refreshed
- Date-Time Bug Fixed
- Refactor UI 

Test Steps
-----------
To enable and build QGC with UTMSP Adapter support using cmake on CLI 
1. Make sure you cloned the repository and updated the submodules before, see chapter Source Code above and switch into the repository folder: 
```
cd qgroundcontrol
```
2. Configure: 
```
cmake -B build -G Ninja -DCMAKE_BUILD_TYPE=Debug -DQGC_UTM_ADAPTER=ON
```
3. Build: 
```
cmake --build build --config Debug
```
4. Run the QGroundcontrol binary
```
./build/QGroundControl
``` 

Checklist:
----------

- [x] [Review Contribution Guidelines](https://github.com/mavlink/qgroundcontrol/blob/master/.github/CONTRIBUTING.md).
- [x] [Review Code of Conduct](https://github.com/mavlink/qgroundcontrol/blob/master/.github/CODE_OF_CONDUCT.md).
- [x] I have tested my changes.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

https://github.com/user-attachments/assets/2d8efc87-9cdb-4466-b47c-0bc2708b21f8

### CODE REVIEW

**Review by bkueng — COMMENTED** [review]
I just have some smaller comments.
Can you also add some screenshots?

And I would add a CI build with `QGC_UTM_ADAPTER` to ensure it keeps building.

File: src/UTMSP/UTMSPAuthorization.h [hunk 1]
```diff
@@ -14,23 +14,22 @@
...
 {
     Q_OBJECT
 public:
-    UTMSPAuthorization();
-    ~UTMSPAuthorization();
+    UTMSPAuthorization(QObject *parent = nullptr);
+    virtual ~UTMSPAuthorization() = default;
 
     const std::string& getOAuth2Token();
+    std::string base64_encode(unsigned char const* bytes_to_encode, unsigned int in_len);
```
[the remark this answers is not in the document]
Comment (rpashchapur) [reply src/UTMSP/UTMSPAuthorization.h]: Removed in [bc38e0f](https://github.com/mavlink/qgroundcontrol/pull/11628/commits/bc38e0fdefd5c3b280eb7a998b5529facaefd5c8)

File: src/UTMSP/UTMSPAuthorization.h [hunk 2]
```diff
@@ -14,23 +14,22 @@
...
-protected:
-    http::request<http::string_body> _request;
-
 private:
-    std::string          _clientID;
-    std::string          _clientSecret;
     bool                 _isValidToken;
+    UTMSPRestInterface   _utmspRestInterface;
+
+    static const std::string base64_chars;
```
[the remark this answers is not in the document]
Comment (rpashchapur) [reply src/UTMSP/UTMSPAuthorization.h]: Removed in [bc38e0f](https://github.com/mavlink/qgroundcontrol/pull/11628/commits/bc38e0fdefd5c3b280eb7a998b5529facaefd5c8)

File: src/UTMSP/UTMSPRestInterface.cpp [hunk 3]
```diff
@@ -10,191 +10,100 @@
...
+    default:
+        qDebug() << "Unsupported HTTP method: " << _currentMethod;
+        return qMakePair(0, "Unsupported HTTP method");
+    }
+
+    if (!reply) return qMakePair(0, "Failed to create network reply");
+
+    QEventLoop loop;
+    connect(reply, &QNetworkReply::finished, &loop, &QEventLoop::quit);
+    loop.exec();
```
[the remark this answers is not in the document]
Comment (rpashchapur) [reply src/UTMSP/UTMSPRestInterface.cpp:98]: Yeah true, this issue I'll resolve in upcoming PR :)

File: src/PlanView/MissionItemMapVisual.qml [hunk 4]
```diff
@@ -35,7 +35,7 @@ Item {
             if (component.status === Component.Error) {
                 console.log("Error loading Qml: ", object.mapVisualQML, component.errorString())
             }
-            _visualItem = component.createObject(map, { "map": _root.map, vehicle: _root.vehicle, 'opacity': Qt.binding(function() { return _root.opacity }), 'interactive': Qt.binding(function() { return _root.interactive }) })
+            _visualItem = component.createObject(map, { "map": _root.map, vehicle: _root.vehicle, 'opacity': 0, 'interactive': Qt.binding(function() { return _root.interactive }) })
```
[the remark this answers is not in the document]
Comment (bkueng) [reply src/PlanView/MissionItemMapVisual.qml:38]: The diff is still here though.

[the remark this answers is not in the document]
Comment (rpashchapur) [reply src/PlanView/MissionItemMapVisual.qml]: Ahh.. Sorry I thought I updated it. Please find the fix in [b49e7d3](https://github.com/mavlink/qgroundcontrol/pull/11628/commits/b49e7d39a0ca00d6eedcb0e6b14d54a101eb3da0)

**Review by bkueng — APPROVED** [review]
Nice work, and thanks for sharing a video

**Pull request merged**